### PR TITLE
feat: using Vault for storing credentials

### PR DESCRIPTION
**Pull request by aureamunoz**

### CODE REVIEW

File: app/src/main/java/io/halkyon/services/BindApplicationService.java [hunk 1]
```diff
@@ -29,6 +31,11 @@ public class BindApplicationService {
     public static final String USERNAME_KEY = "username";
     public static final String PASSWORD_KEY = "password";
 
+    public static final String VAULT_KV_PATH_KEY = "vault-path";
+
+    @Inject
+    VaultKVSecretEngine kvSecretEngine;
```
Comment (cmoulliard) [inline app/src/main/java/io/halkyon/services/BindApplicationService.java:37]: If you hard code the dependency with Vault here, how will it be possible to switch to use another Secret Management system ?

File: app/src/main/java/io/halkyon/services/BindApplicationService.java [hunk 2]
```diff
@@ -66,13 +73,18 @@ private void unMountSecretVolumeEnvInApplication(Claim claim) throws ClusterConn
 
     private void createSecretForApplication(Claim claim, Credential credential, String url)
             throws ClusterConnectException {
+        Map<String, String> vaultSecret = kvSecretEngine.readSecret(credential.vaultKvPath);
```
Comment (cmoulliard) [inline app/src/main/java/io/halkyon/services/BindApplicationService.java:76]: Can you add a if/then block to generate an error, return a proper message when we cannot find for the Path searched the credential ?

**Review by cmoulliard — COMMENTED** [review]
Do you know why the job is failing ? Is it because we dont install/run vault on the port 8200 ?

**Comment by aureamunoz** [comment]
> Do you know why the job is failing ? Is it because we dont install/run vault on the port 8200 ?

Yes, it's normal. The tests are not correctly set up for Vault. It's the next step I will deal with. This is still a draft pull request

File: app/src/main/java/io/halkyon/services/BindApplicationService.java (hunk 1 above)
Comment (aureamunoz) [reply app/src/main/java/io/halkyon/services/BindApplicationService.java:37]: At the moment this is a poc and I was trying to use Vault. For using another management system we need to put in place an abstraction layer to hide it.

**Comment by aureamunoz** [comment]
This is ready for review. I think the rest (set up infra for prod etc) can be done in another PR, I will move forward on it.

**Review by cmoulliard — COMMENTED** [review]
Still no java interface and implement class to decouple the service from Vault. Why ?

**Pull request merged**